src/storage-controller/src/persist_handles.rs

```diff
@@ -583,7 +583,7 @@ impl<T: Timestamp + Lattice + Codec64 + TimestampManipulation> TxnsTableWorker<T
         let handles = ids_handles.into_iter().map(|(_, handle)| handle);
         let res = self.txns.register(register_ts.clone(), handles).await;
         match res {
-            Ok(_ts) => {
+            Ok(tidy) => {
```

Comment:
Wouldn't we still need the `apply_eager_le` for the `!lazy` case? The new register impl only advances the physical upper of the newly registered data shard, but in eager mode we want to be sure to advance all of them before emitting the new_uppers for them all below.

Reply:
Ah ok, yeah that sounds completely right to me. I forgot that `apply_eager_le` advanced the upper of all shards. Fixed.

Reply:
Cool, new impl here lgtm